compiler/rustc_passes/src/dead.rs

```diff
@@ -724,10 +790,58 @@ impl<'tcx> DeadVisitor<'tcx> {
                     );
                     err.note(&msg);
                 }
-                err.emit();
-            });
+                    err.emit();
+                },
+            );
+        }
+    }
+
+    fn warn_dead_fields_and_variants(&self, hir_id: hir::HirId, participle: &str, is_field: bool) {
+        if let Some(parent_node) = self.tcx.hir().find(hir_id)
+            && matches!(
+                parent_node,
+                Node::Item(hir::Item {
+                    kind: hir::ItemKind::Struct(..)
+                        | hir::ItemKind::Union(..)
+                        | hir::ItemKind::Enum(..),
+                    ..
+                })
+            )
+            && let Some(did) = self.tcx.hir().opt_local_def_id(hir_id)
+            && let Some(dead_codes) = self.dead_variants.get(&did)
+        {
+            let mut dead_codes = dead_codes
+                .iter()
+                .filter(|v| v.is_field == is_field)
```

Comment:
Should the `_` prefix checking be done here? Then that check can be removed from `warn_multiple_dead_codes`

Reply:
Previously, `store_dead_field_or_variant` does it, but I moved the checking to `warn_dead_fields_and_variants`.